src/ORM/Table.php

```diff
@@ -2249,20 +2253,34 @@ protected function _saveMany(iterable $entities, $options = []): iterable
                     }
                 });
         } catch (Exception $e) {
-            $cleanup($entities);
+            $cleanupOnFailure($entities);
 
             throw $e;
         }
 
         if ($failed !== null) {
-            $cleanup($entities);
+            $cleanupOnFailure($entities);
 
             throw new PersistenceFailedException($failed, ['saveMany']);
         }
 
+        $cleanupOnSuccess = function (EntityInterface $entity) {
```

Comment:
```suggestion
        $cleanupOnSuccess = function (EntityInterface $entity) use ($cleanupOnSuccess) {
```

Think you need this to fix the lint errors.

Reply:
Didn't help, I shut up the static analyzers instead.

Reply:
I do this with `use (&$cleanupOnSuccess)`

Reply:
Yup, I was missing the `&`.